src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -603,7 +608,7 @@ private Settings buildIndexSettings(IndexMetaData metaData) {
     }
 
     /**
-     * Adds a pending delete for the given index.
+     * Adds a pending delete for the given index shard.
      */
     public void addPendingDelete(Index index, ShardId shardId, Settings settings) {
```

Comment:
yeah I don't like @Nullable it doesn't buy me anything to be honest... I will throw an exception